packages/kit/src/utils/streaming.js

```diff
@@ -23,22 +23,29 @@ function defer() {
  * }}
  */
 export function create_async_iterator() {
-	let deferred = defer();
+	let deferred = [defer()];
 
 	return {
 		iterator: {
 			[Symbol.asyncIterator]() {
 				return {
-					next: () => deferred.promise
+					next: async () => {
+						const next = await deferred[0].promise;
+						if (!next.done) deferred.shift();
+						return next;
+					}
```

Comment:
```suggestion
					next: deferred.shift().promise
```

Reply:
This needs a type cast because TS doesn't know this is always defined, and strictly speaking it's not adhering to the spec that says you can call `next` as much as you want after it's done (it should always return `done: true` in that state).